File: problems/Remove Duplicates from Sorted Array/removeDuplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let lastIdx = 0;

    for (let idx = 1; nums.length > idx; ++idx) {
        if (nums[idx] !== nums[lastIdx]) {
            nums[++lastIdx] = nums[idx];
        }
    }

    return 1 + lastIdx;
};

File: problems/Remove Duplicates from Sorted Array/removeDuplicates.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int result = 1;
        int nonDuplicateIdx = 1;

        for(int idx = 0; nums.length > idx; ++idx) {
            if (0 != idx && nums[idx - 1] != nums[idx]) {
                nums[nonDuplicateIdx++] = nums[idx];
                ++result;
            }
        }

        return result;
    }
}
